internal/provider/tag_resource.go

```diff
@@ -170,7 +174,10 @@ func (r *TagResource) ImportState(ctx context.Context, req resource.ImportStateR
 	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
 }
 
-func (r *TagResource) doTag(ctx context.Context, data *TagResourceModel) (string, error) {
+var tagmu sync.Mutex
+var tags = map[string]map[v1.Hash]struct{}{}
```

Comment:
I'd prefer this as provider state to globals, but not sure how hard that is to plumb?